### ISSUE TITLE
Member Group Picker + ncNodeName

### ISSUE THREAD

**Issue by TRexStark** [leekelleher/umbraco-contentment#376]
### Which Contentment version are you using?

4.6.0

### Which Umbraco version are you using? For example: 10.3.2 - don't just write v10

10.8.3

### Bug summary

First, thank you for such an awesome package!

Using the Contentment Member Group Picker in a native Umbraco Block List Editor (due to the picker having the search option given we have an extensive list).

To populate the Block List item label, usually we would do "{{propertyAlias | ncNodeName}}".
The ncNodeName as we know expects a UDI which contains "umb://..."

1. The Member Group Picker only returns uid in an array object like so ["xxxx-xxxx-...."] (with the dashes)
2. The UDI returned and displayed in the item description shows the full UDI as "umb://member-group/xxxxxxx..." (uid with no dashes).

To be able to populate the label as expected, I followed the instructions from umbraco on how to populate the media item but in conjunction with what the Member Group picker provides:
`{{('umb://membergroup/'+propertyAlias[0].split("-").join("") | ncNodeName)}}`

If you noticed, I called "umb://membergroup" not "member-group" because the api does not have the object with the dash.

### Steps to reproduce

Create a simple Block List item and add contentment Member Group Picker in it and populate the label using my method above.

### Expected result / actual result

1. The UDI should be "umb://membergroup" not "umb://member-group"
2. It will be great if the returned value is the full UDI instead of the uid so we can just do {{propertyAlias | ncNodeName}}
3. This is more of a UX enhancement, but I don't think we need the UDI displayed in the preview? Content editors don't need to see that and it's not relevant to them anyway.

Thank you.

### Do you have Umbraco ModelsBuilder enabled?

- [X] Yes, it is enabled.

**Comment by leekelleher** [comment]
@TRexStark Thanks for reporting this.

It's an interesting one. Firstly, I had thought that the `ncNodeName` filter only handled content/documents, but it turns out that it can support [any Umbraco entity-type](https://github.com/umbraco/Umbraco-CMS/blob/release-10.8.3/src/Umbraco.Web.UI.Client/src/common/filters/nestedcontent.filter.js#L61).

_Warning plenty of links/research coming up..._

At this point, I looked at what I'd coded for `UmbracoMemberGroupDataListSource`, when [the UDI is created (for the description)](https://github.com/leekelleher/umbraco-contentment/blob/4.6.0/src/Umbraco.Community.Contentment/DataEditors/DataList/DataSources/UmbracoMemberGroupDataListSource.cs#L55), it's making reference to [Umbraco's own `Constants.UdiEntityType.MemberGroup` string](https://github.com/umbraco/Umbraco-CMS/blob/release-10.8.3/src/Umbraco.Core/Constants-UdiEntityType.cs#L45), which is `"member-group"` (hyphenated).

So then why wouldn't the `ncNodeName` filter work with it?

Turns out that the API endpoint that is called has the signature [`EntityController.GetById(Guid id, UmbracoEntityTypes type)`](https://github.com/umbraco/Umbraco-CMS/blob/release-10.8.3/src/Umbraco.Web.BackOffice/Controllers/EntityController.cs#L1518), and what is being send over is 2 string parameters, e.g. a Guid string for `id` and `"member-group"` for the `type`. I check what [`UmbracoEntityTypes`](https://github.com/umbraco/Umbraco-CMS/blob/release-10.8.3/src/Umbraco.Core/Models/ContentEditing/UmbracoEntityTypes.cs) is; it's an enum - cool - I see the [`MemberGroup` entry](https://github.com/umbraco/Umbraco-CMS/blob/release-10.8.3/src/Umbraco.Core/Models/ContentEditing/UmbracoEntityTypes.cs#L46). But how can `"member-group"` map to `UmbracoEntityTypes.MemberGroup`? ...and therein lies the issue. Yup, we've uncovered a bug in Umbraco. How to fix it? That's a whole different topic.

**TL;DR, `"member-group"` is the correct UDI entity type name, but there's a bug in Umbraco that doesn't support it.**

---

My advise would be to make your own copy of the [`UmbracoMemberGroupDataListSource`](https://github.com/leekelleher/umbraco-contentment/blob/4.6.0/src/Umbraco.Community.Contentment/DataEditors/DataList/DataSources/UmbracoMemberGroupDataListSource.cs) and set the item value to however works best for you.

```csharp
public IEnumerable<DataListItem> GetItems(Dictionary<string, object> config)
{
    return _memberGroupService
        .GetAll()
        .Select(x => new DataListItem
        {
            Name = x.Name,
            // NOTE: You can't use `Udi.Create()` here,
            // as "membergroup" (without hyphen isn't an Umbraco entity type name)
            Value = $"umb://membergroup/{x.Key:N}",
            Icon = Umbraco.Core.Constants.Icons.MemberGroup,
            Description = "Put whatever you want the description to be here, or null."
        });
}
```

I hope this helps.

**Comment by TRexStark** [comment]
Thanks @leekelleher, that makes sense now ... I didn't dig deeper into the UmbracoEntityType, hence I assumed it was a contentment bug.

Thank you for investigating this, I guess we should transfer this to an umbraco issue, but not sure if they're still fixing v10 bugs like this?
